test/widgets/profile/edit_brith_date_page_test.dart

```diff
@@ -0,0 +1,140 @@
+      expect(birthDateInput, findsOneWidget);
+
+      //check if birthDate TextField is editable
+      await tester.tap(birthDateInput);
+      await tester.pumpAndSettle();
+      expect(find.byType(DatePickerDialog), findsOneWidget);
+    });
+    testWidgets('change birthDate', (WidgetTester tester) async {
+      //load page
+      await pumpMaterial(tester, EditBirthDatePage(profile));
+      await tester.pump();
+
+      //check if birthDate is displayed
+      final Finder birthDateInput = find.text(localeManager.formatDate(profile.birthDate!));
+      expect(birthDateInput, findsOneWidget);
+
+      //tap birthDate TextField
+      await tester.tap(find.byKey(const Key('birthDateInput')));
+      await tester.pumpAndSettle();
+
+      //tap in DatePickerDialog to custom input
+      await tester.tap(find.byIcon(Icons.edit));
+      await tester.pumpAndSettle();
+
+      //enter new birthDate
+      await tester.tap(birthDateInput.last);
+      final String date = localeManager.formatDate(profile.birthDate!.add(const Duration(days: 5)));
+
+      //leave DatePickerDialog
+      await tester.enterText(birthDateInput.last, date);
+      await tester.tap(find.text('OK'));
+      await tester.pumpAndSettle();
+
+      //check if new birthDate is displayed
+      expect(find.text(date), findsOneWidget);
+    });
+    testWidgets('clear Button', (WidgetTester tester) async {
+      //load page
+      await pumpMaterial(tester, EditBirthDatePage(profile));
+      await tester.pump();
+
+      //check if birthDate is displayed
+      expect(find.text(localeManager.formatDate(profile.birthDate!)), findsOneWidget);
+
+      //check if clear Button is displayed
+      final Finder clearButton = find.byKey(const Key('clearButton'));
+      expect(clearButton, findsOneWidget);
+
+      //check if birthDate is cleared
+      await tester.tap(clearButton);
+      await tester.pump();
+      expect(find.text(localeManager.formatDate(profile.birthDate!)), findsNothing);
+    });
+    testWidgets('save Button', (WidgetTester tester) async {
+      //sign in
+      SupabaseManager.supabaseClient.auth.signInWithPassword(
+        email: email,
+        password: authId,
+      );
+
+      //load page
+      await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+      await tester.pump();
+
+      //check if EditBirthDatePage is displayed
+      await tester
+          .tap(find.descendant(of: find.byKey(const Key('age')), matching: find.byKey(const Key('editButton'))));
+      await tester.pumpAndSettle();
+      expect(find.byType(EditBirthDatePage), findsOneWidget);
+
+      //check if save Button is displayed
+      expect(find.byKey(const Key('saveButton')), findsOneWidget);
+
+      //tap save Button
+      await tester.tap(find.byKey(const Key('saveButton')));
+      await tester.pumpAndSettle();
+
+      //check if ProfilePage is displayed
+      expect(find.byType(ProfilePage), findsOneWidget);
```

Comment:
this should test that the actually correct new age is shown now

Reply:
we can't test if the changes from edit are shown in profile, since it fetches the new values from Supabase. So it would just be me telling it in any situation that it should change the value.
The only thing I could do is that I check if an Supabase update was called after the tap on savebutton.
And then on Profile page if loadProfile is called again.

Reply:
Yes please just test that there is a supabase call to patch the data